### ISSUE TITLE
Pixie Event Header Word 0, Finish Code

### ISSUE THREAD

**Issue by ksmith0** [pixie16/paass#131]
The unpacker is grabbing the `eventLength` (bits 17-28), `virtualChannel` (bit 29), `saturatedBit` (bit 30) and `pileupBit` (bit 31) from the incorrect portion of the first header word for modern firmwares (at least Feb 2, 2016 and newer). These modern firmwares have event length (bits 17-30) and finish code (bit 31).
For example, after Mar 8, 2016:
![image](https://cloud.githubusercontent.com/assets/7395803/20283361/af3d5a7c-aa86-11e6-9cec-5846c66a4bbf.png)

The excerpt below containing the relevant code is from the following method in Unpacker.cpp
```c++
int Unpacker::ReadBuffer(unsigned int *buf, unsigned long &bufLen)
```

Lines 235-243 of Unpacker.cpp 
```c++
unsigned int chanNum      = (buf[0] & 0x0000000F);
unsigned int slotNum      = (buf[0] & 0x000000F0) >> 4;
unsigned int crateNum     = (buf[0] & 0x00000F00) >> 8;
unsigned int headerLength = (buf[0] & 0x0001F000) >> 12;
unsigned int eventLength  = (buf[0] & 0x1FFE0000) >> 17;

currentEvt->virtualChannel = ((buf[0] & 0x20000000) != 0);
currentEvt->saturatedBit   = ((buf[0] & 0x40000000) != 0);
currentEvt->pileupBit      = ((buf[0] & 0x80000000) != 0);
```

**Comment by ksmith0** [comment]
This may affect those who are currently using PAASS for analysis and have long traces as the `eventLength` is being truncated to a smaller value (unpacker max of 4095 words, loss of 12288 out of 16383 words). It may also affect those  who are interested in the `virtualChannel`, `saturatedBit` or `pileupBit` as they are being pulled from the bit of the `eventLength` and `finishCode`. Finally, the Finish Code is not transparently available for those who would like to use it.

**Comment by spaulaus** [comment]
This is sort of a duplicate of issue #8 . Users should be knowledgeable on the firmware used to collect the data. How should we as developer's handle this?

**Comment by ksmith0** [comment]
I believe we need to add a firmware argument for unpacking and have a method for each firmware. We may be able to make some default that is reasonable most of the time or just default to the most recent firmware.

**Comment by spaulaus** [comment]
I think defaulting to the most recent is reasonable. In the case of utkscan this could be handled with the configuration file. What about scope/skeleton? Command line arguments?

**Comment by ksmith0** [comment]
Command line arguments seem reasonable, perhaps a toggle inside the program as well.

**Comment by spaulaus** [comment]
This has been fixed as of pull request #156. We now decode nine different data formats, but still need to include formats older than 02/15/2014.
